### ISSUE TITLE
I try fine-tuning model with ADE20K 25K image data loss does not fall

### ISSUE THREAD

**Issue by wuhongsheng** [open-mmlab/mmclassification#1004]
推荐使用英语模板 General question，以便你的问题帮助更多人。

### 首先确认以下内容

- 我已经查询了相关的 issue，但没有找到需要的帮助。
- 我已经阅读了相关文档，但仍不知道如何解决。

### 描述你遇到的问题
I try fine-tuning model with ADE20K 25K image data loss does not fall

### 相关信息

1. `pip list | grep "mmcv\|mmcls\|^torch"` 命令的输出
   \[填写这里\]
2. 如果你修改了，或者使用了新的配置文件，请在这里写明

```python
mmcls                         0.23.2                       /content/mmclassification
mmcv                          1.6.1
torch                         1.12.1+cu113
torchaudio                    0.12.1+cu113
torchsummary                  1.5.1
torchtext                     0.13.1
torchvision                   0.13.1+cu113
```

3. 如果你是在训练过程中遇到的问题，请填写完整的训练日志和报错信息
   #@title 默认标题文本
# 载入已经存在的配置文件
```
from mmcv import Config
from mmcls.utils import auto_select_device

#config_file = 'configs/mobilenet_v2/mobilenet-v2_8xb32_in1k.py'
#checkpoint_file = 'https://download.openmmlab.com/mmclassification/v0/mobilenet_v2/mobilenet_v2_batch256_imagenet_20200708-3b2dc3af.pth'
classes = ['0', '180', '270', '90']  # The category names of your dataset



#config_file = 'configs/resnet/resnet50_8xb32_in1k.py'
#checkpoint_file = 'https://download.openmmlab.com/mmclassification/v0/resnet/resnet50_8xb32_in1k_20210831-ea4938fc.pth'
config_file = 'configs/resnet/resnet18_8xb32_in1k.py'
checkpoint_file = 'https://download.openmmlab.com/mmclassification/v0/resnet/resnet18_8xb32_in1k_20210831-fbbb1da6.pth'
# vit
#config_file = 'configs/vision_transformer/vit-base-p32_ft-64xb64_in1k-384.py'
#checkpoint_file = 'https://download.openmmlab.com/mmclassification/v0/vit/finetune/vit-base-p32_in21k-pre-3rdparty_ft-64xb64_in1k-384_20210928-9cea8599.pth'

#cifar10
#config_file = 'configs/resnet/resnet18_8xb16_cifar10.py'
#checkpoint_file = 'https://download.openmmlab.com/mmclassification/v0/resnet/resnet18_batch256_imagenet_20200708-34ab8f90.pth'

cfg = Config.fromfile(config_file)
cfg.device = auto_select_device()

# 修改模型分类头中的类别数目
cfg.model.head.num_classes = 4
cfg.model.head.topk = (1, )

# 加载预训练权重
#cfg.model.backbone.frozen_stages=2
cfg.model.backbone.init_cfg = dict(type='Pretrained', checkpoint=checkpoint_file, prefix='backbone')


# 根据你的电脑情况设置 sample size 和 workers 
cfg.data.samples_per_gpu = 1
cfg.data.workers_per_gpu = 1

# 指定训练集路径
cfg.data.train.data_prefix = '/content/drive/MyDrive/datasets/ADE20K_OUTDOORS/images/train'
#cfg.data.train.classes = '/content/drive/MyDrive/datasets/ADE20K_OUTDOORS/images/classes.txt'
cfg.data.train.classes = classes
# 指定验证集路径
cfg.data.val.data_prefix = '/content/drive/MyDrive/datasets/ADE20K_OUTDOORS/images/val'
cfg.data.val.ann_file = '/content/drive/MyDrive/datasets/ADE20K_OUTDOORS/images/val.txt'
cfg.data.val.classes = classes
#cfg.data.val.classes = '/content/drive/MyDrive/datasets/ADE20K_OUTDOORS/images/classes.txt'

# 指定测试集路径
cfg.data.test.data_prefix = '/content/drive/MyDrive/datasets/ADE20K_OUTDOORS/images/test'
cfg.data.test.ann_file = '/content/drive/MyDrive/datasets/ADE20K_OUTDOORS/images/test.txt'
cfg.data.test.classes = '/content/drive/MyDrive/datasets/ADE20K_OUTDOORS/images/classes.txt'

# 设定数据集归一化参数
normalize_cfg = dict(mean=[124.508, 116.050, 106.438], std=[58.577, 57.310, 57.437], to_rgb=True)
# img_norm_cfg = dict(
#     mean=[125.307, 122.961, 113.8575],
#     std=[51.5865, 50.847, 51.255],
#     to_rgb=False,
# )

# cfg.data.train.pipeline[3] = normalize_cfg
# cfg.data.val.pipeline[3] = normalize_cfg
# cfg.data.test.pipeline[3] = normalize_cfg


train_pipeline = [
    dict(type='LoadImageFromFile'),
    dict(type='RandomResizedCrop', size=224),
    #dict(type='RandomFlip', flip_prob=0.5, direction='horizontal'),
    dict(type='Normalize', **normalize_cfg),
    dict(type='ImageToTensor', keys=['img']),
    dict(type='ToTensor', keys=['gt_label']),
    dict(type='Collect', keys=['img', 'gt_label'])
]

cfg.data.train.pipeline = train_pipeline
# cfg.data.train.pipeline = [
#     dict(type='LoadImageFromFile'),
#     dict(type='Resize', size=256),
#     #dict(type='CenterCrop', crop_size=224),
#     dict(type='Normalize', **normalize_cfg)
# ]
# cfg.data.val.pipeline = [
#     dict(type='LoadImageFromFile'),
#     dict(type='Resize', size=256),
#     #dict(type='CenterCrop', crop_size=224),
#     dict(type='Normalize', **normalize_cfg)
# ]

# cfg.data.test.pipeline = [
#     dict(type='LoadImageFromFile'),
#     dict(type='Resize', size=256),
#     #dict(type='CenterCrop', crop_size=224),
#     dict(type='Normalize', **normalize_cfg)
# ]


# 修改评价指标选项
cfg.evaluation['metric_options']={'topk': (1, )}

# 设置优化器
cfg.optimizer = dict(type='SGD', lr=0.01, momentum=0.9, weight_decay=0.0001)
cfg.optimizer_config = dict(grad_clip=None)

# 设置学习率策略
#cfg.lr_config = dict(policy='step', step=1, gamma=0.1)
cfg.lr_config = dict(policy='step', step=[15])

cfg.runner = dict(type='EpochBasedRunner', max_epochs=5)

# 设置工作目录以保存模型和日志
cfg.work_dir = './work_dirs/orientation_dataset'

# 设置每 10 个训练批次输出一次日志
cfg.log_config.interval = 10

# 设置随机种子，并启用 cudnn 确定性选项以保证结果的可重复性
from mmcls.apis import set_random_seed
cfg.seed = 0
set_random_seed(0, deterministic=True)
cfg.gpu_ids = range(1)
```
4. 如果你对 `mmcls` 文件夹下的代码做了其他相关的修改，请在这里写明
   \[填写这里\]

**Comment by Ezra-Yu** [comment]
1. use [this tool](https://mmclassification.readthedocs.io/en/latest/tools/visualization.html#pipeline-visualization) to check your dataset
2. try a bigger batch_size and image size
3. try another model like `ResNet` 
4. try to use smaller `lr` and other scheduler
```
cfg.optimizer = dict(type='SGD', lr=0.001, momentum=0.9, weight_decay=1e-5)

lr_config = dict(
    policy='CosineAnnealing',
    min_lr=0,
    warmup='linear',
    warmup_iters=2,
    warmup_ratio=0.01,
    warmup_by_epoch=True)

cfg.runner = dict(type='EpochBasedRunner', max_epochs=10)

```

**Comment by wuhongsheng** [comment]
> 1. use [this tool](https://mmclassification.readthedocs.io/en/latest/tools/visualization.html#pipeline-visualization) to check your dataset
> 2. try a bigger batch_size and image size
> 3. try another model like `ResNet`
> 4. try to use smaller `lr` and other scheduler
> 
> ```
> cfg.optimizer = dict(type='SGD', lr=0.001, momentum=0.9, weight_decay=1e-5)
> 
> lr_config = dict(
>     policy='CosineAnnealing',
>     min_lr=0,
>     warmup='linear',
>     warmup_iters=2,
>     warmup_ratio=0.01,
>     warmup_by_epoch=True)
> 
> cfg.runner = dict(type='EpochBasedRunner', max_epochs=10)
> ```

Thank you, I found a bug in my data processing
